oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/TokenHandler.java

```diff
@@ -209,4 +207,30 @@ public APIGatewayProxyResponseEvent handleRequest(
                                     200, tokenResponse.toJSONObject().toJSONString());
                         });
     }
+
+    private Subject getSubjectByEmailAndClient(String email, ClientRegistry client) {
+        UserProfile userProfile = dynamoService.getUserProfileByEmail(email);
+
+        if (client.getSubjectType().equalsIgnoreCase("public")) {
+            return new Subject(userProfile.getPublicSubjectID());
+        } else {
+            List<ClientSubject> clientSubjects = userProfile.getClientSubjects();
+            if (client.getSectorIdentifierUri() != null) {
+                for (ClientSubject clientSubject : clientSubjects) {
```

Comment:
Minor point, you could use the streaming API `filter(...).findFirst()` rather than a for loop here.